### PR TITLE
GH-2572 Treat ConfirmType.SIMPLE same as ConfirmType.CORRELATED

### PR DESCRIPTION
**Pull request by bjoernhaeuser**
GH-2572 Do not close channel when `ConfirmType.SIMPLE` is used for publisher confirmations.

See #2571 (discussion) and #2572 (issue) for reference.

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java [hunk 1]
```diff
@@ -2344,9 +2344,6 @@ public <T> T invoke(OperationsCallback<T> action, @Nullable com.rabbitmq.client.
 				if (channel == null) {
 					throw new IllegalStateException("Connection returned a null channel");
 				}
-				if (!connectionFactory.isPublisherConfirms()) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java:2347]: Why remove altogether?
The discussion was about condition improvement:
```
if (!connectionFactory.isPublisherConfirms() && !connectionFactory.isSimplePublisherConfirms()) {
```

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java (hunk 1 above)
Comment (bjoernhaeuser) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java:2347]: Thanks for taking the time to review this!

I went through the discussion and thought the code and came to (maybe faulty) conclusion that closing the channel never makes sense for the `invoke` method. Especially for me this totally unexpected and not mentioned in the docs anywhere. 

Please let me know which way you would like to see this proceed!

Thanks.

Comment (artembilan) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java:2347]: I believe the close is not required when we are in publish-confirm mode.
Therefore that condition and another PR.
Sorry, I don't see anywhere in discussion that we were going to remove this condition altogether.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java [hunk 2]
```diff
@@ -2344,7 +2344,7 @@ public <T> T invoke(OperationsCallback<T> action, @Nullable com.rabbitmq.client.
...
+				if (!connectionFactory.isPublisherConfirms() && !connectionFactory.isSimplePublisherConfirms()) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java:2347]: Isn't the change in #2651 more precise and accurate?

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java (hunk 2 above)
Comment (bjoernhaeuser) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java]: Sorry, this was still a bit WIP. Added this change here as well and added also test for this. Hope this covers this completely.

**Review by artembilan — CHANGES_REQUESTED** [review]
Looks like there is failing existing test since we have changed condition, I guess:
```
RabbitTemplateTests > testPublisherConnWithInvokeAndSimpleConfirmations() FAILED
    org.opentest4j.AssertionFailedError: 
    Expecting value to be false but was true
        at app//org.springframework.amqp.rabbit.core.RabbitTemplateTests.testPublisherConnWithInvokeAndSimpleConfirmations(RabbitTemplateTests.java:631)
```

**Comment by artembilan** [comment]
I had to revert the fix to our original discussion: https://github.com/spring-projects/spring-amqp/commit/2bb4f297931ac39a49d320bd6c81ddf4797c5562.
Apparently there are use-cases when listener is added to the `PublisherCallbackChannel` which makes sense only for the `ConfirmType.CORRELATED` mode.

**Pull request merged**